lib/typing_env.ml

```diff
@@ -158,18 +158,21 @@ let expand_tconstr ~typing_env ~path ~args =
       match td.Types.type_manifest with
       | None -> None
       | Some type_expr ->
-          Some (Ctype.apply typing_env td.Types.type_params type_expr args))
+          let expanded_expr =
+            Ctype.apply typing_env td.Types.type_params type_expr args
+          in
+          Some (expanded_expr, td))
 
-let fully_expand_tconstr ~typing_env ~path ~args =
-  let rec aux last path args =
+let expansion_lst ~typing_env ~type_expr ~path ~args =
```

Comment:
It'd be good if we could avoid passing the type_expr here, I think it ends up messing with the return type more than anything an clutters a bit the function when we don't really need it.

We can see it as returning the expansion chain from the tconstr made of `path` and `args`. The caller could eventually use this and the type_expr it used to get the path and args to build another list, more suited to their own need. In other words, I think injecting the type_expr here is the caller code leaking into the implementation of this function.